main.cc

```diff
@@ -1268,6 +1268,11 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
 
             db::sstables_format_selector sst_format_selector(db);
 
+            api::set_format_selector(ctx, sst_format_selector).get();
+            auto stop_format_seletor_api = defer_verbose_shutdown("sstables format selector API", [&ctx] {
+                api::unset_format_selector(ctx).get();
+            });
+
             supervisor::notify("starting system keyspace");
```

Comment:
Like, encapsulate it in an object with two methods, and have a  one-liner to do this.

Reply:
Pretty much the same applies to

```
    sharded<Foo> x;
    co_await = x.start();
    auto stop_foo = defer([&x] { co_await x.stop(); }); // co_await is not possible in defer though :(
```

which could be

```
    auto start_sharded_and_get_deferred_stop(sharded<T>& x) {
        co_await = x.start();
        return defer([&x] { co_await x.stop(); });
    }
    sharded<Foo> x;
    auto stop_foo = co_await start_sharded_and_get_deferred_stop(x);
```